quickwit-metastore/src/metastore/mod.rs

```diff
@@ -193,7 +193,15 @@ pub struct MetadataSet {
 /// running search queries can complete.
 #[cfg_attr(any(test, feature = "testsuite"), mockall::automock)]
 #[async_trait]
-pub trait Metastore: Send + Sync + 'static {
+pub trait Metastore: Check + Send + Sync + 'static {
+    /// Check if the given index is in this metastore
+    async fn check_index(&self, index_id: &str) -> anyhow::Result<()> {
+        if let Err(e) = self.index_metadata(index_id).await {
+            anyhow::bail!(e.to_string())
```

Comment:
`if ... { anyhow::bail!(e.to_string()) }` instead of  
`self.index_metadata(index_id).await?;` is meant to reduce the error message length?

Reply:
Exactly. On the other hand, perhaps that additional info is useful, so I'll use your suggestion instead.